src/concurrent_map/warp/insert.cuh

```diff
@@ -93,4 +93,92 @@ GpuSlabHashContext<KeyT, ValueT, SlabHashTypeT::ConcurrentMap>::insertPair(
     }
     last_work_queue = work_queue;
   }
+}
+
+/*
+ * each thread inserts a unique key (and its value) into the hash table
+ * it is assumed all threads within a warp are present and collaborating with
+ * each other with a warp-cooperative work sharing (WCWS) strategy.
+ */
+template <typename KeyT, typename ValueT>
+__device__ __forceinline__ void
+GpuSlabHashContext<KeyT, ValueT, SlabHashTypeT::ConcurrentMap>::insertPairUnique(
```

Comment:
So this function insert a key-value pair only if the key is not found in the data structure. So, the first instance always remain in the data structure. Another variation is that we can have something like `replacePair` where the last instance always survive, though it maintains uniqueness. Just wanted to raise the discussion here so that we know which one we're going to need exactly. Also, it will be probably better to mention this as a comment, higher in the code.

Reply:
yeah I agree, that could be useful. I added an issue for this so we don't forget about it.